### PR TITLE
HP-339

### PR DESCRIPTION
**Pull request by ahinchliff**
Add convert logic to SDK

### CODE REVIEW

**Review by rellfy — APPROVED** [review]

File: src/types/Convert.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      await this.setTokenAddressToType();
+    }
+
+    const { data } = await this.client.get("/swap/v1/quote", {
+      params: {
+        buyToken,
+        sellToken,
+        sellAmount,
+        buyAmount,
+        feeRecipient: "0x19835c8126d1c56c83A746DfDc9738Bb4a987B9B",
```
Comment (bitcoinbrisbane) [inline src/types/Convert.ts:71]: Who’s address is this?

File: src/types/Convert.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      console.error("Failed to fetch token list. User will be charged the highest fee");
+    }
+  };
+
+  private getFees = (tokenA: string, tokenB: string) => {
+    const typeAType = this.tokenAddressToType?.[tokenA];
+    const typeBType = this.tokenAddressToType?.[tokenB];
+
+    if (!typeAType || !typeBType) {
+      return "0.003";
```
Comment (bitcoinbrisbane) [inline src/types/Convert.ts:105]: Why are theses strings?

File: tests/suites/convert.spec.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    const networkTwoTokens = await secondConvert.getTokens();
+
+    expect(networkOneTokens.length !== networkTwoTokens.length).toBeTruthy();
+  });
+
+  it("Should fetch price data for a trade", async () => {
+    const priceResult = await convert.getQuote(
+      "ETH",
+      "0x4f96fe3b7a6cf9725f59d353f723c1bdb64ca6aa",
+      ethers.utils.parseEther("1").toString(),
```
Comment (bitcoinbrisbane) [inline tests/suites/convert.spec.ts:37]: Why cast to strings?

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
Why are we not using numbers or big numbers?

**Review by xiangzhe0812 — APPROVED** [review]

File: src/types/Convert.ts (hunk 1 above)
Comment (ahinchliff) [reply src/types/Convert.ts:71]: This is seahounds and is the same for all chains.

File: src/types/Convert.ts (hunk 2 above)
Comment (ahinchliff) [reply src/types/Convert.ts:105]: Because that it what the 0x api expects

File: tests/suites/convert.spec.ts (hunk 3 above)
Comment (ahinchliff) [reply tests/suites/convert.spec.ts:37]: So developers using our skd dont need to use BigNumber

File: src/types/Convert.ts (hunk 1 above)
Comment (rellfy) [reply src/types/Convert.ts:71]: You could add it to `config,json` although we won't need it once we have the convert wrapper contract

File: src/types/Convert.ts (hunk 2 above)
Comment (bitcoinbrisbane) [reply src/types/Convert.ts:105]: Ok

File: tests/suites/convert.spec.ts (hunk 3 above)
Comment (bitcoinbrisbane) [reply tests/suites/convert.spec.ts:37]: Big number is acceptable if not encouraged in crypto

Comment (rellfy) [reply tests/suites/convert.spec.ts:37]: I also agree we should only use BN whenever possible -- I missed this when reviewing

Comment (ahinchliff) [reply tests/suites/convert.spec.ts:37]: sure, but different projects use different Big Number packages. We dont want to force them to use a package they dont use.

Comment (ahinchliff) [reply tests/suites/convert.spec.ts]: Done

**Review by rellfy — APPROVED** [review]

**Pull request merged**